Core/include/Acts/Utilities/VectorHelpers.hpp

```diff
@@ -211,5 +212,50 @@ inline auto makeVector4(const Eigen::MatrixBase<vector3_t>& vec3,
   return vec4;
 }
 
+// Computes the signed angle \in [-pi, pi] between a 2D vector vec and a 2D reference vector refVec
+// @note: the vectors should lie in the same plane
+// @note: for an explanation of why this computation works see
+//      https://stackoverflow.com/questions/5188561/
+// @note: in future Eigen releases, the nominator can be replaced by refVec.cross(vec), see
+//      https://gitlab.com/libeigen/eigen/-/merge_requests/1098
+template <typename Derived>
+inline typename Derived::Scalar signedAngle(
+    const Eigen::MatrixBase<Derived>& vec,
+    const Eigen::MatrixBase<Derived>& refVec) {
+  return (refVec[0] * vec[1] - refVec[1] * vec[0]) / refVec.dot(vec);
+}
+
+// Decomposes a 3D vector into a transverse and a longitudinal component with
+// respect to the z-direction
+template <typename Derived>
+inline Eigen::Matrix<typename Derived::Scalar, 2, 1> decomposeTransverseZ(
+    const Eigen::MatrixBase<Derived>& vec) {
+  Eigen::Matrix<typename Derived::Scalar, 2, 1> decomposedVec{perp(vec.head(2)),
+                                                              vec[2]};
+  return decomposedVec;
+}
+
+// Subtracts phi and theta
+// minPhiTheta and subPhiTheta should be 2D vectors containing phi in the 0th
+// entry and theta in the first entry
+// @note: the prefix "min" stands for minuend and the prefix "sub" for subtrahend
+// @note: Simply subtracting phis is problematic due to the periodicity of phi
+// @note: Simply subtracting thetas is problematic around the poles of the unit sphere (i.e., at theta close to 0 or pi)
+template <typename Derived>
+inline Eigen::Matrix<typename Derived::Scalar, 2, 1> subtractPhiTheta(
```

Comment:
since this breaks down at the poles I am not sure how much sense it makes to generalize it that far. I think operations near the pole should fail as the derivatives will be meaningless anyways

Reply:
Ok, I will remove these functions then!